Prog/Master School Update Through 2017-18.sas

```diff
@@ -0,0 +1,226 @@
+/**************************************************************************
+ Program:  Master School Update Through 2017.sas
+ Library:  Schools
+ Project:  Urban-Greater DC
+ Author:   Rob Pitingolo
+ Created:  01/15/2019
+ Version:  SAS 9.4
+ Environment:  Local Windows session (desktop)
+ 
+ Description:  Takes most recent master school file and adds 3 newest years
+			   of data to bring it up to date as of the 2017-18 school year. 
+
+ Modifications:
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( Schools )
+libname old 'L:\Libraries\Schools\Data\Master school file\14_15_data';
+libname enroll 'L:\Libraries\Schools\Data\Enrollment';
+
+/* Use old MSF for data prior to 2014 */
+data schools_00_14;
+	set old.Msf_final_00_14_flip;
+
+	if school_name = "CLOSED";
+	status = "CLOSED";
+
+	/* Keeping only the geoblk2000 geo var to add others back on later */
+	keep ui_id master_school_name dcps year school_name geoblk2000 aud status; 
+run;
+
+
+/* Use geographic data from 2015/16 file as key for 16/17 and 17/18 */
+data school_locations;
+	set enroll.Enroll_15_16;
+	ui_id_n = ui_id;
+	keep ui_id_n x y ;
+run;
+
+proc sort data = school_locations; by ui_id_n; run;
+
+
+/* List closed schools as of 2014 school year */
+data closed_schools;
+	set Schools_00_14;
+	if year = 2014;
+	if school_name = "CLOSED";
+	status = "CLOSED";
+	ui_id_n = ui_id+0;
+	keep ui_id_n master_school_name dcps status;
+run;
+
+proc sort data = closed_schools; by ui_id_n; run;
+
+
+/*Use Enrollment files for school directory for 2015, 2016 and 2017 */
+proc sort data = enroll.Enroll_15_16 
+	out = schools_15_16_in (rename = (ui_id=ui_id_n)); 
+	by ui_id; 
+run;
+
+proc sort data = enroll.Enroll_16_17 
```

Comment:
**Enroll_16_17** and **Enroll_17_18** have *ui_id_n* values that are duplicated across different schools. This is not consistent with the structure of the previous MFS, where ID numbers are unique for each school. The duplicate ID values are

- *ui_id_n*=1021300 : "BRIDGES PCS" and "BRIGHTWOOD EC"
- *ui_id_n*=2100302 : "CESAR CHAVEZ PCS FOR PUBLIC POLICY - PARKSIDE HS" and "CESAR CHAVEZ PCS FOR PUBLIC POLICY - PARKSIDE MS"
- *ui_id_n*=2102403 : "DC PREP PCS - ANACOSTIA CAMPUS" and "DC PREP. PCS- BENNING MS"

These are all new schools that do not appear in the 2000-14 MFS. The different schools need to be assigned unique ID numbers.

Reply:
The first one appears to be a typo that I simply corrected to the right number. The other two appear to be new campuses of existing charter schools. I used the guidance in the instructions to create new ui_ids for them.